tests/integration/sync/test_sync_build_in_source.py

```diff
@@ -0,0 +1,141 @@
+import os
+import shutil
+import pytest
+import logging
+import json
+from unittest import skip
+from parameterized import parameterized
+
+from tests.integration.sync.sync_integ_base import SyncIntegBase
+from tests.integration.sync.test_sync_code import TestSyncCodeBase
+from tests.testing_utils import run_command_with_input
+from samcli.lib.utils.resources import AWS_LAMBDA_FUNCTION
+
+LOG = logging.getLogger(__name__)
+
+
+@skip("Building in source option is not exposed yet. Stop skipping once it is.")
+class TestSyncInfra_BuildInSource_Makefile(SyncIntegBase):
+    dependency_layer = False
+
+    def setUp(self):
+        super().setUp()
+        self.new_files_in_source = [
+            self.test_data_path.joinpath(
+                "code/before/makefile_function_create_new_file/file-created-from-makefile-function"
+            ),
+            self.test_data_path.joinpath("code/before/makefile_layer_create_new_file/file-created-from-makefile-layer"),
+        ]
+
+    def tearDown(self):
+        super().tearDown()
+        for path in self.new_files_in_source:
+            if os.path.isfile(path):
+                os.remove(path)
+
+    @parameterized.expand(
+        [
+            (True, True),  # build in source
+            (False, False),  # don't build in source
+            (None, False),  # use default for workflow (don't build in source)
```

Comment:
why None? seems odd since everything else is bool.

Reply:
In Lambda Builders, each workflow has a different directory they use to build in by default (some use scratch, some artifacts, some source dir). So there's the option to build in source (build-in-source is True), not build in source (build-in-source is False), or use the workflow's default (build-in-source is None). We could use a False value to indicate that we should use the workflow's default, but it would be weird for someone to say "don't build in source" (setting it to False), and then the default for the workflow is to build in source so it would indeed end up building in source